client/rpc-spec-v2/src/chain_head/tests.rs

```diff
@@ -1317,3 +1326,87 @@ async fn follow_report_multiple_pruned_block() {
 	});
 	assert_eq!(event, expected);
 }
+
+#[tokio::test]
+async fn follow_finalized_before_new_block() {
```

Comment:
To make sure I follow this properly, I think it is:

- first send "finalized: block 1"; this should trigger NewBlock, BestBlock then Finalized for that block
- then send "new block: block 1"; this should basically be ignored since we've sent up to finalized above
- then "new block: block 2" should trigger a NewBlock and BestBlock for block 2 since it's now the new and de facto best block seen to date.

So yup, I think that all makes sense to me :)

Reply:
Yep, that's it! Have added a few comments on the test to help us in the future in case we have to revisit this test, thanks!